### DISCUSSION TITLE
Transparency detection in palette PNGs

### DISCUSSION THREAD

**Discussion by Seishiin** [SixLabors/ImageSharp#2208] [Q&A] [answered]
Is it possible to detect if a palette-based PNG has transparency without having to decode the whole image?
```cs
var imageInfo = Image.Identify("example.png");
var metadata = imageInfo.Metadata.GetPngMetadata();
var colorType = metadata.ColorType; // PngColorType.Palette
var transparency = metadata.HasTransparency; // false
var bpp = imageInfo.PixelType.BitsPerPixel; // 8
```
Am I missing something or is it just not possible?
[example.zip](https://github.com/SixLabors/ImageSharp/files/9378215/example.zip)

**Comment by JimBobSquarePants** [comment, marked as answer]
You're looking for [`Image.Identify`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_Identify_SixLabors_ImageSharp_Configuration_System_IO_Stream_)

**Reply by Seishiin** [reply]
I already tried that, but the image I provided returns `false` for `HasTransparency`.

**Reply by JimBobSquarePants** [reply]
Ah right.... 

I think you may have found a bug. Looking at our decoder code we only assign true to the property if the color type is Rgb or GrayScale.

Can you please raise an issue?

**Reply by Seishiin** [reply]
Will do.
Thanks.
